### PR TITLE
Moved LST and Event classes to ALPAKA_ACCELERATOR_NAMESPACE

### PR DESCRIPTION
**Pull request by ariostas**
I moved the `LST` and `Event` classes to the `ALPAKA_ACCELERATOR_NAMESPACE::lst` namespace. I ended up getting `nvcc` to work by being extra careful. I tried to make the minimum changes possible, but it still ended up being quite a bit. There's some function declarations that ended up being pretty long. Maybe later we can revisit this and introduce some `using` statements to make them cleaner. Or if you think that keeping the templated classes would be better for now, then we can just drop this PR and fix it later.

### CODE REVIEW

**Comment by slava77** [comment]
time to resolve the conflicts in Event.dev.cc

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 1]
```diff
@@ -2,9 +2,12 @@
...
 #include "Event.h"
 
-using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+using Device = ALPAKA_ACCELERATOR_NAMESPACE::Device;
+using Queue = ALPAKA_ACCELERATOR_NAMESPACE::Queue;
+using Acc1D = ALPAKA_ACCELERATOR_NAMESPACE::Acc1D;
+using Acc3D = ALPAKA_ACCELERATOR_NAMESPACE::Acc3D;
 
-void lst::Event<Acc3D>::initSync(bool verbose) {
+void ALPAKA_ACCELERATOR_NAMESPACE::lst::Event::initSync(bool verbose) {
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:10]: I don't see this pattern in other dev.cc files, `namespace ALPAKA_ACCELERATOR_NAMESPACE {` is used instead at the top of the file

File: RecoTracker/LSTCore/interface/alpaka/Constants.h [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 namespace lst {
```
Comment (slava77) [inline RecoTracker/LSTCore/interface/alpaka/Constants.h:12]: would it make sense to put everything to `namespace ALPAKA_ACCELERATOR_NAMESPACE {` ?

**Review by slava77 — COMMENTED** [review]
isn't this still partial and the request was to move all code in /alpaka/ to `ALPAKA_ACCELERATOR_NAMESPACE` ?

**Comment by ariostas** [comment]
I'll finish it tomorrow morning. Is that fine?

**Comment by slava77** [comment]
> I'll finish it tomorrow morning. Is that fine?

OK

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc (hunk 1 above)
Comment (ariostas) [reply RecoTracker/LSTCore/src/alpaka/Event.dev.cc]: Okay, I switched to `using namespace ALPAKA_ACCELERATOR_NAMESPACE::lst;` at the top

File: RecoTracker/LSTCore/interface/alpaka/Constants.h (hunk 2 above)
Comment (ariostas) [reply RecoTracker/LSTCore/interface/alpaka/Constants.h]: Yeah, I think that makes sense

**Comment by ariostas** [comment]
> isn't this still partial and the request was to move all code in /alpaka/ to `ALPAKA_ACCELERATOR_NAMESPACE`?

I'm not sure if that's what they asked for, but it definitely makes things easier.


I made all these changes. It ended up being a very big diff, so we could postpone this for a post-merge cleanup if we're unsure of something.

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 3]
```diff
@@ -951,16 +924,16 @@ void lst::Event<Acc3D>::createPixelQuintuplets() {
...
   auto* connectedPixelIndex_host = connectedPixelIndex_host_buf.data();
 
-  int pixelIndexOffsetPos =
-      pixelMapping_.connectedPixelsIndex[size_superbins - 1] + pixelMapping_.connectedPixelsSizes[size_superbins - 1];
-  int pixelIndexOffsetNeg = pixelMapping_.connectedPixelsIndexPos[size_superbins - 1] +
-                            pixelMapping_.connectedPixelsSizesPos[size_superbins - 1] + pixelIndexOffsetPos;
+  int pixelIndexOffsetPos = pixelMapping_.connectedPixelsIndex[::size_superbins - 1] +
+                            pixelMapping_.connectedPixelsSizes[::size_superbins - 1];
+  int pixelIndexOffsetNeg = pixelMapping_.connectedPixelsIndexPos[::size_superbins - 1] +
+                            pixelMapping_.connectedPixelsSizesPos[::size_superbins - 1] + pixelIndexOffsetPos;
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:930]: is this still needed considering `    using ::lst::size_superbins;` in RecoTracker/LSTCore/interface/alpaka/Constants.h ?

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 4]
```diff
@@ -1061,7 +1031,7 @@ void lst::Event<Acc3D>::addMiniDoubletsToEventExplicit() {
 
   for (unsigned int i = 0; i < nLowerModules_; i++) {
     if (!(nMDsCPU[i] == 0 or module_hitRanges[i * 2] == -1)) {
-      if (module_subdets[i] == Barrel) {
+      if (module_subdets[i] == ::lst::Barrel) {
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:1034]: is it worth to add in RecoTracker/LSTCore/interface/alpaka/Constants.h ?

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h [hunk 5]
```diff
@@ -347,16 +348,19 @@ namespace lst {
...
+                                                              : kMiniMulsPtScaleEndcap[iL] * 3.f / ptCut);
+    const bool isTilted =
+        modulesInGPU.subdets[moduleIndex] == ::lst::Barrel and modulesInGPU.sides[moduleIndex] != ::lst::Center;
     //the lower module is sent in irrespective of its layer type. We need to fetch the drdz properly
 
     float drdz;
     if (isTilted) {
-      if (modulesInGPU.moduleType[moduleIndex] == PS and modulesInGPU.moduleLayerType[moduleIndex] == Strip) {
+      if (modulesInGPU.moduleType[moduleIndex] == ::lst::PS and
+          modulesInGPU.moduleLayerType[moduleIndex] == ::lst::Strip) {
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/MiniDoublet.h:363]: also to add in RecoTracker/LSTCore/interface/alpaka/Constants.h ?

**Review by slava77 — APPROVED** [review]
some cleanup of `::lst` cases could be done in a followup.
It seems to me that some definitions are not even needed in the global `::lst` and could live in RecoTracker/LSTCore/interface/alpaka/Constants.h

**Pull request merged**